### ISSUE TITLE
Memory leak in `[dataOperation URLSession:session task:task didCompleteWithError:error];`

### ISSUE THREAD

**Issue by liaodalin19903** [rs/SDWebImage#1744]
### New Issue Checklist

`[dataOperation URLSession:session task:task didCompleteWithError:error];`  // this line in `- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error` cause memory leak.

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios
 Platform Version        | e.g.
 SDWebImage Version      | e.g. 3.8.1
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 8.1
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

In `intrusment`:

![image](https://cloud.githubusercontent.com/assets/12727582/21085977/214f7500-c053-11e6-95d8-008295a6cad1.png)

**Comment by liaodalin19903** [comment]
@Rob Ryan, Rob, bro, How do you think this memory leak to solve?

**Comment by Cylix** [comment]
Any news about this leak?
I got the same and the leak is particularly severe (currently the thing that leaks the most in our app).
Best

**Comment by ashishprasadgit** [comment]
perhaps this can help
[[SDWebImageDownloader sharedDownloader] setShouldDecompressImages:false]
Decompressing Image is good for performance but consumes a lot of memory.

**Comment by dreampiggy** [comment]
Could you please try the lastest version ? Our if you use custom downloader instance, manually call `invalidateAndCancel` on the URLSession instance if you do not need it.

**Comment by dreampiggy** [comment]
This issue is not active for long time. Feel free to reopen it if you're still interested in this issue.
